### PR TITLE
[mlrun-kit] Add values file to install the kit with MinIO instead of NFS

### PR DESCRIPTION
**Pull request by eliyahu77**
### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove irrelevant fields.]
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### Description:
Add minio support standalone  mode 

tested on: 
k8s version -> v1.21.5
helm version -> v3.6.3

checks:
- minio persistency host path use with pv,pvc support only standalone mode
- jupyter-notebook include with persistency local host pv and pvc for  standalone mode

there few mode to install minio chart , in values-minio.yaml the mode is standalone for minimal installation .

to install chart for advance installation   we can change the  mode value to distributed , and enable persistence , in addition need to consider change resource requests, replicas, in service type .

the service type that use on values-minio.yaml is NodePort the expose the follwoing ports:
 * minio api -> 30080
 * minio console -> 30090

mlrun integrate with minio and communicate via service of k8s .

### CODE REVIEW

File: stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml [hunk 1]
```diff
@@ -20,7 +20,6 @@ spec:
         runAsGroup: 100
         fsGroup: 100
       initContainers:
-{{- if .Values.jupyterNotebook.persistency.enabled }}
```
Comment (quaark) [inline stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml:23]: Could you please explain why this is now done also when the persistency isn't enabled?

File: stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml [hunk 2]
```diff
@@ -51,17 +49,19 @@ spec:
...
         - name: MLRUN_PVC_MOUNT
           value: {{ printf "%s:/home/jovyan/data" (include "mlrun-kit.shared-persistency-pvc.fullname" .) }}
+{{- else }}
+          value: {{ printf "%s:/home/jovyan/data" "jupyter-pvc" }}
+{{- end }}
         - name: CHOWN_HOME
           value: "yes"
         - name: CHOWN_HOME_OPTS
           value: "-R"
-{{- if .Values.jupyterNotebook.persistency.enabled }}
```
Comment (quaark) [inline stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml:60]: same ^

File: stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml [hunk 3]
```diff
@@ -72,10 +72,12 @@ spec:
         - --NotebookApp.token=''
         - --NotebookApp.password=''
         - --NotebookApp.default_url="/lab"
-{{- if .Values.jupyterNotebook.persistency.enabled }}
```
Comment (quaark) [inline stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml:75]: same ^

File: stable/mlrun-kit/templates/jupyter-notebook/pv.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  name: jupyter-pv
+  labels:
+    app.kubernetes.io/name: {{ template "mlrun-kit.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+spec:
+  storageClassName: "jupyter-host-path"
+  capacity:
+    storage: 2Gi
```
Comment (quaark) [inline stable/mlrun-kit/templates/jupyter-notebook/pv.yaml:11]: This should be a value that can be configured in the values file

File: stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+spec:
+  storageClassName: "jupyter-host-path"
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 2Gi
```
Comment (quaark) [inline stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml:14]: same ^

File: stable/mlrun-kit/templates/minio-persistency/pv.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  name: minio-pv
+  labels:
+    app.kubernetes.io/name: {{ template "mlrun-kit.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+spec:
+  storageClassName: "minio-host-path"
+  capacity:
+    storage: 2Gi
```
Comment (quaark) [inline stable/mlrun-kit/templates/minio-persistency/pv.yaml:11]: same ^

File: stable/mlrun-kit/templates/minio-persistency/pvc.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+spec:
+  storageClassName: "minio-host-path"
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 2Gi
```
Comment (quaark) [inline stable/mlrun-kit/templates/minio-persistency/pvc.yaml:14]: same ^

**Review by quaark — CHANGES_REQUESTED** [review]

File: stable/mlrun-kit/README.md [hunk 8]
```diff
@@ -11,6 +11,7 @@ The Open source MLRun kit chart includes the following stack:
 * Jupyter - https://github.com/jupyter/notebook (+MLRun integrated)
 * NFS - https://github.com/kubernetes-retired/external-storage/tree/master/nfs
 * MPI Operator - https://github.com/kubeflow/mpi-operator
+* Minio - https://github.com/minio/minio/tree/master/helm/minio
```
Comment (Hedingber) [inline stable/mlrun-kit/README.md]:
```suggestion
* MinIO - https://github.com/minio/minio/tree/master/helm/minio
```

Comment (Hedingber) [reply stable/mlrun-kit/README.md]: Please fix in all places

File: stable/mlrun-kit/templates/jupyter-notebook/pv.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.jupyterPersistencyStandalone.pvc.create -}}
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: jupyter-pv
+  labels:
+    app.kubernetes.io/name: {{ template "mlrun-kit.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
Comment (Hedingber) [inline stable/mlrun-kit/templates/jupyter-notebook/pv.yaml]:
```suggestion
    {{- include "mlrun-kit.jupyter.labels" . | nindent 4 }}
```

File: stable/mlrun-kit/templates/jupyter-notebook/pv.yaml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.jupyterPersistencyStandalone.pvc.create -}}
```
Comment (Hedingber) [inline stable/mlrun-kit/templates/jupyter-notebook/pv.yaml:1]: No need to add a new `jupyterPersistencyStandalone` block
You already have `jupyterNotebook.persistency.pvc.create` note that this file is creating a PV and not PVC, so should have a different flag for that

File: stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.jupyterPersistencyStandalone.pv.create -}}
```
Comment (Hedingber) [inline stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml:1]: `jupyterNotebook.persistency.pvc.create`

File: stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: jupyter-pvc
+  labels:
+    app.kubernetes.io/name: {{ template "mlrun-kit.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
Comment (Hedingber) [inline stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml]: Same as for pv, use labels from helpers

File: stable/mlrun-kit/templates/minio-persistency/pv.yaml [hunk 13]
```diff
@@ -0,0 +1,21 @@
+{{- if .Values.minioPersistencyStandalone.pvc.create -}}
```
Comment (Hedingber) [inline stable/mlrun-kit/templates/minio-persistency/pv.yaml:1]:
```suggestion
{{- if .Values.minioPersistency.pv.create -}}
```

File: stable/mlrun-kit/templates/minio-persistency/pv.yaml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: minio-pv
+  labels:
+    app.kubernetes.io/name: {{ template "mlrun-kit.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
Comment (Hedingber) [inline stable/mlrun-kit/templates/minio-persistency/pv.yaml]: Add an helper for minio and use it here and in the PVC

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: stable/mlrun-kit/templates/jupyter-notebook/pv.yaml (hunk 10 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/jupyter-notebook/pv.yaml:1]: pv,pvc  in separate block in order not to case  brake compatibility in original mlrun values.yaml

File: stable/mlrun-kit/templates/minio-persistency/pv.yaml (hunk 13 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/minio-persistency/pv.yaml:1]: pv,pvc  create local volume on host path , working only on standalone Minio  mode

File: stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml (hunk 1 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml:23]: when persistency isn't enabled , by default  the chart will create pv,pvc on local volume with host path

File: stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml (hunk 2 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml:60]: same ^

File: stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml (hunk 3 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/jupyter-notebook/deployment.yaml:75]: same ^

File: stable/mlrun-kit/templates/jupyter-notebook/pv.yaml (hunk 4 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/jupyter-notebook/pv.yaml:11]: this is the pv that create the local volume

File: stable/mlrun-kit/templates/minio-persistency/pv.yaml (hunk 6 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/minio-persistency/pv.yaml:11]: this is the pv that create the local volume

File: stable/mlrun-kit/templates/minio-persistency/pvc.yaml (hunk 7 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/minio-persistency/pvc.yaml:14]: this is the pv that create the local volume

File: stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml (hunk 11 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml:1]: pv,pvc in separate block in order not to case brake compatibility in original mlrun values.yaml

File: stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml (hunk 12 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/jupyter-notebook/pvc.yaml]: done

File: stable/mlrun-kit/templates/minio-persistency/pv.yaml (hunk 14 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/minio-persistency/pv.yaml]: done

**Pull request merged**